### PR TITLE
Add some additional functionality

### PR DESCRIPTION
**Pull request by brenca**
Added hotkeys
- Ctrl/Meta + Scroll _to Zoom_
- Ctrl/Meta + PageDown/PageUp _to always jump a whole page_
- Ctrl/Meta + Alt + w/p _to force fit-to-**w**idth and fit-to-**p**age_

Added url parameters
- view: _fitW/fitP_ - uses fit-to-page or fit-to-width on load (standard open pdf parameter AFAIK)
- toolbar: _2_ - added the 2 value that makes the toolbar visible, but doesn't include empty space for it at the beginning of the document

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -49,4 +49,8 @@ var viewer;
 
     createBrowserApi(opts).then(initViewer);
   });
+
+  global.pdfLoaded = new Promise(function(resolve, reject) {
```
Comment (deepak1556) [inline main.js]: This seems redundant, we already get notified when the viewer is ready with `setLoadCallback`. https://github.com/electron/electron/blob/master/spec/chromium-spec.js#L1004

File: pdf.js [hunk 2]
```diff
@@ -258,6 +262,44 @@ function PDFViewer(browserApi) {
 
   // Request translated strings.
   cr.sendWithPromise('getStrings').then(this.handleStrings_.bind(this));
+
+  // Setup global functions for scripting from electron
+  global.fitToWidth = function() {
```
Comment (deepak1556) [inline pdf.js]: We have the pdf viewer instance available for scripting with `window.viewer`, its better to use it in Electron as required for testing, can help reduce the changes to this fork. Thoughts ?

**Review by deepak1556 — COMMENTED** [review]
LGTM, with minor changes. Thanks for this @brenca !

File: pdf.js (hunk 2 above)
Comment (brenca) [reply pdf.js]: Sorry, somehow I missed that we already have a viewer reference :(

**Comment by brenca** [comment]
Fixed! 👀

File: pdf.js [hunk 3]
```diff
@@ -495,6 +518,7 @@ PDFViewer.prototype = {
       return;
     if (this.loadCallback_)
       this.loadCallback_(this.loadState_ == LoadState.SUCCESS);
+    window.dispatchEvent(new Event('pdf-loaded'))
```
Comment (deepak1556) [inline pdf.js]: the event can be removed.

**Review by deepak1556 — APPROVED** [review]

File: pdf.js (hunk 3 above)
Comment (brenca) [reply pdf.js]: In the specs, we are using the page-title-updated event, which is not really a solid way of dealing with the loading I think, but I have found that at dom-ready the viewer is not initialized yet, that's why I left this event in there. This way, we don't have to wait for some extra event to happen on the webContents after load, we can just listen to this on the window object any time. I'd argue that this makes things easier, and is a one line change with not so much of maintenance impact.

Comment (deepak1556) [reply pdf.js]: Sounds good but the event and `loadCallback_` are doing the same job and we already use the latter, so it seems redundant to have both. The `page-title-updated` event is just used to set the `loadCallback_`.

**Comment by brenca** [comment]
@kevinsawicki :eyes:

**Comment by kevinsawicki** [comment]
Thanks @brenca, pulling into Electron via https://github.com/electron/electron/pull/9892

**Pull request merged**